src/core/thread/mle.cpp

```diff
@@ -2660,6 +2660,12 @@ void Mle::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessageIn
 #endif
 #endif // OPENTHREAD_FTD
 
+#if OPENTHREAD_MTD && OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+    case kCommandTimeSync:
+        Get<TimeSync>().HandleTimeSyncMessage(rxInfo.mMessage);
```

Comment:
Should we be calling `MleRouter::HandleTimeSync()` instead?
https://github.com/openthread/openthread/blob/d099d786201712449d8fff4f489941c87cfeaebe/src/core/thread/mle_router.cpp#L3899

In particular, there's extra logic there to check for valid neighbor state and logging. If so, maybe we should move `HandleTimeSync()` to this class.

Reply:
Since MleRouter don't exist in MTD, I will try to move HandleSyncTime in mle class and test it.